packages/components/src/combobox-control/index.tsx

```diff
@@ -107,6 +109,10 @@ function ComboboxControl( {
 	};
 
 	const handleArrowNavigation = ( offset = 1 ) => {
+		if ( ! selectedSuggestion ) {
+			return;
+		}
+
 		const index = matchingSuggestions.indexOf( selectedSuggestion );
```

Comment:
If `selectedSuggestion` is falsy, this can't look for it in `matchingSuggestions`